### PR TITLE
Serveral fixes

### LINKED ISSUE #6339: Colonists disappeared, blocks can't be interacted with and can't load a backup
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.4
- MineColonies Version: minecolonies-0.13.547-ALPHA-universal
- Structurize Version: structurize-0.13.104-ALPHA-universal
- Related Mods and their Versions: none

## Expected behavior
Being able to restore my world by loading a backup, whether i do it ingame using commands or manually from the minecolonies folder

## Actual behavior
Colonist are all missing, can't interact with any of the building blocks and trying to do what other threats with similar issues have done doesn't work, loading a backup ingame gives a succesfully loaded backup message but doesn't have any effect, doing it from the minecolonies folder doesn't have any effect either, i might be doing that las one wrong, i copy the .dat file from one of the .rars and replace the one already in the folder, tell me if i have to do it in another way, and when trying to use the hall building it says there is already a colony nearby. Using some commands like delete it says ''colony with ID 1 could not be found in this dimension!'' even tho, the hall building clearly detects there is a colony and it doesn't let me access it

## Steps to reproduce the issue

I was trying to restore my Nether dimension, so i deleted the DIM-1 folder from the minecraft world, but, before doing that i made a backup of the whole world just i case, when i opened the world without the DIM-1 folder all the problems i mention appeared, but removing and replacing that world with the backup i had just make was of no use either.

1. Make a backup of the world
2. Delete the DIM-1 from the world in the saves folder
3. Open the game
4. Replace the world in the saves folder with the backup previously made 
5. Open the game again to find out it's still broken

### Logs

 

- latest.log:  https://gist.github.com/DarthPepo/3798e57062de05ba570df89895df370b
- crashlog: 

### Notes

This is my first issue here, my apologies if i've done something wrong 

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #6406: deleted colony gets automatically restored as i restart my forge server
### Prerequisites 

- [ x ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [ x ] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [ x ] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version:
- MineColonies Version: minecolonies-0.13.566-ALPHA-universal.jar
- Structurize Version: structurize-0.13.104-ALPHA-universal
- Related Mods and their Versions:

## Expected behavior
 The deleted colony should stay deleted and i should be able to build a new colony (when i try to build a new colony it just says i should try it somewhere else, at this point the old one does not exist anymore or does not show up in colony list until i restart my server and it gets automatically restored for some reason, while trying to build a new one the townhall block also says im not near an existing colony, neither am i the owner of one)

## Actual behavior
 I succesfully delete an existing colony but it automatically gets restored as soon as i restart my server. I can't build a new colony near it even right after deleting the old one (does not show up in colony list until i restart my server and it gets automatically restored) 

## Steps to reproduce the issue

1.  Delete an existing colony and try to build a new one (which does not work on my end)
2.  Restart forge server
3.  Deleted colony gets restored for some reason 

### Logs

- latest.log:  https://gist.github.com/nicoo1708/f9d17575e1ce64d1f39999c3d39cb8db
- crashlog: 

### Notes
 

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #6436: miner requests ladders and voids them when fulfilled
- [x ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [ x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [ x] I made sure that **this issue is not a duplicate of any existing issue.**

- Minecraft Version: 1.15.2
- MineColonies Version: -0.13.562
- Structurize Version: 0.13.105
- Related Mods and their Versions:

my miner wasn't accepting the ladders he was requesting. they would go into his inventory (either by clicking the 'fulfill' button, or directly placing them in the inventory, or given by courier), however shortly after that they would slowly start to disappear - then number decreasing about 3 or 4 per second. they didn't appear in any other inventory nearby ('mine' block, or racks)

i have a darkoak mine level 4, and the cobble block at the top of the ladder was missing a ladder (maybe a barbarian broken it, who knows). replacing the ladder at the top (on the cobble block) fixed the issue.

image

### LINKED ISSUE #6474: Pirate ship spawning and destroying Hospital building
### Prerequisites 

- [ ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version. 
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context  

- Minecraft Version: 1.16.4
- MineColonies Version: 0.13.566
- Structurize Version: 0.13.104
- Related Mods and their Versions: N/A

## Expected behavior

I expect the pirate ship not to spawn inside of a built building, and possibly be a certain amount of blocks away.

## Actual behavior

Pirate Ship spawned inside the area of my hospital, destroying blocks on the edges of my building

image

## Steps to reproduce the issue

Unfortunately the steps to reproduce this error are minimal.
1. Build a large building near the edge of the water
2. wait for pirates to spawn into building and destroy blocks never to be found again

### Logs

- latest.log: 
- crashlog: 

### Notes

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #6526: Pathfinding problem with double doors
### Prerequisites

- [x ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [ ] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x ] I made sure that **this issue is not a duplicate of any existing issue.**

- Not actually runing latest structurize as the latest MineColonies version specificaly requires an older version of Structurize to launch.
- Could not complet "Check wiki" as the link only gives "Bad Gateway" message.

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: 0.13.610-ALPHA
- Structurize Version: 0.13.113-ALPHA
- Related Mods and their Versions: A few smaller mods, can provide compleat list if deamed relevant.

## Expected behavior
When a citisen is trying to pass thorugh a double door, they should open the door actually in front of them (or both of them).

## Actual behavior
Often a citisen will stand directly infront of one of the doors and repetedly opening and closing the adjacent door resulting in them not getting through as the door they should open remain closed. Se video for example: https://youtu.be/IKdcRutbH3I

## Steps to reproduce the issue

1. Have buildings with double doors.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #6576: Fence Connection Check not running after builder completes structure
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.4
- MineColonies Version: 0.13.557-ALPHA
- Structurize Version: 0.13.104-ALPHA
- Related Mods and their Versions: Forge 35.1.35

## Expected behavior
When the builder finishes a building, all of the fences should be updated so they connect with appropriate solid blocks.

## Actual behavior
The builder finishes a structure, and the fences are not updated leaving gaps. Saving to menu and reloading will update the blocks as well as right clicking on them manually, but that seems to not be intended.

## Steps to reproduce the issue
1. Start a colony
2. Build a builder's hut that uses fences (I used Medieval Oak)
3. Observe the missing connections upon completion

### Logs

I trimmed it a bit to have the relevant section.
https://gist.github.com/KKomrade/3b9430d3757d4c05d046932495d60ca1

### Notes

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #6577: Recipe UI closes after each add instead of remaining open until done
### Prerequisites 

- [ ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x ] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [ x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 36.0.42
- MineColonies Version: 0.13.622
- Structurize Version: 0.13.135
- Related Mods and their Versions:

## Expected behavior

Recipe UI should remain open while each recipe gets added as it has in the past.  Having to right-click on the block, click on Add Recipe, click Done and then repeat the same steps is just way too time consuming/frustrating.  Some huts have 10's of recipes to manually add in and this just prolongs the agony of doing so taking away from the fun part of the builds.
## Actual behavior

Something changed recently and now it closes after clicking Done every time.

## Steps to reproduce the issue

I upgraded to 0.13.622 and we added a number of huts/buildings and then we started adding recipes only to find out it now takes much longer to do so.  I was on the Blacksmith adding its recipes and the list is pretty long.
1. 
2. 
3. 

### Logs

- latest.log: 
- crashlog: 

### Notes

Given we are making multiple colonies on our servers, what would be more helpful than manually adding recipes over and over for the same buildings either multiple times in one colony or many colonies, is to have a command or server file setting that can support "require recipe" or "not require recipe" that we can use.  Could extend this further by specifying the colony ID associated with the setting.
---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by someaddons**
Closes #6577
Closes #6526
Closes #6474
Closes #6436
Closes #6406
Closes #6576
Closes #6339

# Changes proposed in this pull request:
Mark citizen manager dirty on citizen death
Increase base raidable range of buildings
Fix miner's voiding ladders by placing air instead of a ladderblock
Fix ToggleAbleAI not considering all possible obstacles, it ignored those who were on the next block but did not block movement to next's next.
Citizens now prefer paths where they do not have to use many toggleable blocks.
Fix some scrolls crashing on getting the colony after shrinking its itemstack to zero.
Fix backups on windows filesystem, those were broken as the name there couldnt contain ":". Instead of using res locations as names, we now distribute colony data backup files into dimension folders.
Fix our util setblockstate, which had an issue with not always syncing changes to the client
Fix closing recipe UI after adding one recipe

Review please

### CODE REVIEW

File: src/api/java/com/minecolonies/api/entity/pathfinding/PathingOptions.java [hunk 1]
```diff
@@ -130,4 +135,10 @@ public PathingOptions withRailExitCost(final double railExitCost)
         railsExitCost = railExitCost;
         return this;
     }
+
+    public PathingOptions withToggleCost(final double toggleCost)
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/entity/pathfinding/PathingOptions.java:139]: no javadoc? =D

Comment (someaddons) [reply src/api/java/com/minecolonies/api/entity/pathfinding/PathingOptions.java:139]: others dont have either :P

**Comment by Raycoms** [comment]
:shipit:

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java [hunk 2]
```diff
@@ -975,7 +976,7 @@ private void setBlockFromInventory(@NotNull final BlockPos location, final Block
         {
             getInventory().extractItem(slot, 1, false);
             //Flag 1+2 is needed for updates
```
Comment (Mekle001) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java:978]: This comment doesn't apply anymore, does it?

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java:978]: ah ye no longer needed

**Review by marchermans — APPROVED** [review]
This looks good.

@someaddons Could you port the toggleable stuff into PVI as well?

**Comment by someaddons** [comment]
ye gonna add it there too

**Pull request merged**